src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -230,13 +230,22 @@ private function addFormSection(ArrayNodeDefinition $rootNode, callable $enableI
                     ->{$enableIfStandalone('symfony/form', Form::class)}()
                     ->children()
                         ->arrayNode('csrf_protection')
-                            ->treatFalseLike(['enabled' => false])
-                            ->treatTrueLike(['enabled' => true])
-                            ->treatNullLike(['enabled' => true])
+                            ->beforeNormalization()
+                                ->always(static fn ($v) => match (\gettype($v)) {
+                                    'string' => ['enabled' => null, 'header_name' => $v],
```

Comment:
This makes it easier to turn on double-submit validation. This snippet does it (and I'd like to propose it for the default recipe v7.2):
```yaml
framework:
    form:
        csrf_protection: x-csrf-token
```